packages/components/src/SvgIcon/SvgIcon.tsx

```diff
@@ -0,0 +1,102 @@
+// @flow
+import * as React from "react";
+import cx from "classnames";
+import styles from "./SvgIcon.module.css";
+
+type BaseProps = {
+  /**
+   * The SVG Color, expects a valid css color (hex, rgb, etc.). You can use
+   * Recommendation: if `currentColor` isn't sufficient, use color tokens from
+   * `import Tokens from '@chanzuckerberg/eds-tokens/json/variables.json';`
+   */
+  color?: string;
+  /**
+   * Either a fragment of svg elements (g, path, circle, etc.), or a single
+   * svg elment
+   */
+  children: React.ReactNode;
+  /**
+   * Width/Height string (px, rem, em, vh, etc.)
+   * Recommendation: use line-height tokens from
+   * `import Tokens from '@chanzuckerberg/eds-tokens/json/variables.json';`
+   */
+  size?: string;
+  /**
+   * Classname for additional custom styling in context.
+   */
+  className?: string;
+  /**
+   * The role of the icon. Should be 'presentation' for icons that don't provide
+   * any information on their own - for example, when an icon doesn't add any
+   * additional information to page text. Should be 'img' for icons that stand
+   * alone and have no other accompanying information - these will require you
+   * to provide `title` text that provides meaning to assistive technology.
+   */
+  role: "presentation" | "img";
+  /**
+   * Boolean, if true, the element will be displayed as a block, otherwise
+   * default is inline like images
+   */
+  block?: boolean;
+};
+
+type PresentationProps = BaseProps & {
+  role: "presentation";
+};
+
+type ImgProps = BaseProps & {
+  role: "img";
+  /**
+   * Accessible title that will be read out with screen readers.
+   */
+  title: string;
+};
+
+export type SvgIconProps = PresentationProps | ImgProps;
+
+interface SvgStyle extends React.CSSProperties {
+  "--svg-icon-size"?: string;
+}
+
+const SvgIcon = React.forwardRef<SVGElement, SvgIconProps>(
```

Comment:
This is the one currently-failing typescript error.

Error output:
```
src/SvgIcon/utils/createSvgIcon.tsx(22,8): error TS2322: Type '{ children: ReactNode; color?: string | undefined; size?: string | undefined; className?: string | undefined; role: "img" | "presentation"; block?: boolean | undefined; ref: any; }' is not assignable to type '(IntrinsicAttributes & BaseProps & { role: "presentation"; } & RefAttributes<SVGElement>) | (IntrinsicAttributes & BaseProps & { ...; } & RefAttributes<...>)'.
  Property 'title' is missing in type '{ children: ReactNode; color?: string | undefined; size?: string | undefined; className?: string | undefined; role: "img" | "presentation"; block?: boolean | undefined; ref: any; }' but required in type '{ role: "img"; title: string; }'.
```

Reply:
If I add `title?: string;` to `BaseProps`, I still get and error over the `role` values being different:
```
src/SvgIcon/utils/createSvgIcon.tsx(22,8): error TS2322: Type '{ children: ReactNode; title?: string | undefined; color?: string | undefined; size?: string | undefined; className?: string | undefined; role: "img" | "presentation"; block?: boolean | undefined; ref: any; }' is not assignable to type '(IntrinsicAttributes & BaseProps & { role: "presentation"; } & RefAttributes<SVGElement>) | (IntrinsicAttributes & BaseProps & { ...; } & RefAttributes<...>)'.
  Type '{ children: ReactNode; title?: string | undefined; color?: string | undefined; size?: string | undefined; className?: string | undefined; role: "img" | "presentation"; block?: boolean | undefined; ref: any; }' is not assignable to type '{ role: "img"; title: string; }'.
    Types of property 'role' are incompatible.
      Type '"img" | "presentation"' is not assignable to type '"img"'.
        Type '"presentation"' is not assignable to type '"img"'.
```

Reply:
I feel like I have to have a wrapper that accepts either prop set and returns one of two components depending on the `role` value.

Reply:
I found [this comment](https://github.com/FB-PLP/traject/pull/5761#discussion_r595636765) from @ahuth from when he added this for MaterialIcon and was hoping it had more context around what we should do in Typescript 😅 

Would introducing a `switch` statement into `createSvgIcon` based on the `role` help?

Reply:
So this works for me:
```
  const Component = (
    props: SvgIconProps,
    ref: React.ForwardedRef<SVGSVGElement>,
  ) => {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const { role: _, ...rest } = props;

    return props.role === "img" ? (
      <SvgIcon ref={ref} role={props.role} title={props.title} {...rest}>
        {path}
      </SvgIcon>
    ) : (
      <SvgIcon ref={ref} role={props.role} {...rest}>
        {path}
      </SvgIcon>
    );
  };
```

but using `IconProps` (after it goes through $Diff or Omit) doesn't work with the same code 🤔  other things that did not work (still threw the 'title' not a property error):
```
const {role, title} = props;
```

```
const {role} = props;

return role === "img" ? <SvgIcon role={role} title={props.title}>
```

Reply:
I think this is hitting a limitation of `Pick`, which is how both [Omit is implemented](https://github.com/microsoft/TypeScript/blob/7c4c0a396203adb9ce5a989b61a04ac25657246e/src/lib/es5.d.ts#L1496)  and [$Diff is implemented](https://github.com/piotrwitek/utility-types/blob/df2502ef504c4ba8bd9de81a45baef112b7921d0/src/utility-types.ts#L50).

Haven't been able to find any official reference to this, but I believe it doesn't "distribute" over a union like we'd expect.

So we have something like

```ts
type IconProps = PresentationProps | ImgProps;
```

and we expect

```ts
Omit<IconProps, 'children'>
```

to essentially be equivalent to

```ts
Omit<PresentationProps, 'children'> | Omit<ImgProps, 'children'>
```

But that doesn't seem to be the case. Not exactly sure what it is equivalent to.

Might be related to any of these (although it's kind of hard to tell. I find these issues/posts hard to read)
- https://github.com/microsoft/TypeScript/issues/12959
- https://stackoverflow.com/questions/57103834/typescript-omit-a-property-from-all-interfaces-in-a-union-but-keep-the-union-s
- https://davidgomes.com/pick-omit-over-union-types-in-typescript/

Reply:
One fix would be to not try to omit `children`, and discard any value for it in `createSvgIcon`.

That would mean someone could pass `children` to an icon, which would promptly get ignored. Of course we'd like a type error there, but maybe it's unlikely to be an issue in this repo.

Reply:
Another idea: get rid of `createSvgIcon` completely and [implement icons like we do in Along](https://github.com/chanzuckerberg/along/blob/62990d05130e55cbf7af2a7c52814160ee594999/client/src/ui/Icons/PDF.tsx).

Basically looks something like

```ts
// SomeIcon.tsx

export default function SomeIcon(props: Props) {
  return (
    <SvgIcon {...props}>
      <Circle />
    </SvgIcon>
  );
}
```

Since there should only be a few icons in here, the `createSvgIcon` abstraction may be unnecessary.

Reply:
I think we do want to move more of the icons over though. We're just not yet because we're about to change the icon set